### PR TITLE
Todo api

### PR DESCRIPTION
**Pull request by akkee**
Added 3 methods to deal with todos, only for the actions provided by the [official API](https://docs.gitlab.com/ce/api/todos.html).

  * `todos`: Gets a list of all or filtered todos
  * `mark_todo_as_done`: Marks a todo as done
  * `mark_all_todos_as_done`: Marks all todos as done

### CODE REVIEW

**Comment by akkee** [comment]
Can this be merged @asedge ? Any changes or additions ?

File: lib/gitlab/client/todos.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Gitlab::Client
+  # Defines methods related to todos
+  # @see https://docs.gitlab.com/ce/api/todos.html 
+  module Todos
+    # Gets a list of todos.
+    #
+    # @example
+    #   Gitlab.todos
+    #   Gitlab.todos(action: 'assigned')
+    #   Gitlab.todos(state: 'pending')
```
Comment (asedge) [inline lib/gitlab/client/todos.rb]: I know it's generally not idiomatic Ruby to include the curly braces but we generate the CLI/Shell help from the YARD so could you include them here?  For the CLI/Shell we require hashes to be passed as valid YAML.

**Comment by akkee** [comment]
Changes done.

**Comment by asedge** [comment]
👍 Thanks!

**Pull request merged**